starter/src/main/java/com/example/starter/base/osgi/VaadinMode.java

```diff
@@ -17,12 +19,11 @@
     String PREFIX_ = "servlet.init.";
 
     /**
-     * {@code compatibilityMode} service property.
+     * {@code frontendDirectory} service property.
      *
-     * @return the compatibilityMode initial parameter value
+     * @return the frontendDirectory initial parameter value
      * @see org.osgi.service.http.whiteboard.HttpWhiteboardConstants.HTTP_WHITEBOARD_SERVLET_INIT_PARAM_PREFIX
-     * @see com.vaadin.flow.server.InitParameters.SERVLET_PARAMETER_COMPATIBILITY_MODE
      */
-    boolean compatibilityMode() default false;
+    String frontendDirectory() default "${project.basedir}/" + FrontendUtils.FRONTEND;
```

Comment:
compat mode should be removed, production mode should be used instead.

`frontendDirectory`  is extremely bad example for the property: it's used only for frontend resources and this never used in OSGi since frontend resource should be already handled by the maven plugin at the time of application execution .
I'm not even sure that  `frontendDirectory` property is used with this name anyhow in the web back dev server initializer: at least I don't see such property name in the `InitParameters` class.

Reply:
Yes, agree. This is related to a frontend part and shouldn't be here.